src/multihash.rs

```diff
@@ -259,6 +275,37 @@ where
     Ok(())
 }
 
+/// Writes the multihash to a byte slice.
+#[cfg(not(feature = "std"))]
+pub fn write_multihash<'a>(
+    w: &'a mut &'a mut [u8],
+    code: u64,
+    size: u8,
+    digest: &[u8],
+) -> Result<(), Error> {
+    use unsigned_varint::encode as varint_encode;
+
+    let mut code_buf = varint_encode::u64_buffer();
+    let code = varint_encode::u64(code, &mut code_buf);
+
+    let mut size_buf = varint_encode::u8_buffer();
+    let size = varint_encode::u8(size, &mut size_buf);
+
+    let result_size = size.len() + code.len() + digest.len();
+    if result_size > w.len() {
+        return Err(Error::InvalidSize(result_size as u64)); // not the right error.
```

Comment:
Perhaps it would make sense to introduce a non-std `Io` error, which could be used here? I'm not sure myself, perhaps it even warrants a completely new error.